bin/generate_sessions.py

```diff
@@ -0,0 +1,153 @@
+Requires `TOKEN` environment variable (pretalx API token)
+"""
+from argparse import ArgumentParser
+import csv
+import json
+import os
+from pathlib import Path
+import random
+import re
+from urllib.parse import urljoin
+
+import requests
+
+BASE_PATH = Path(__file__).parent.parent
+
+
+def api_get(path):
+    api_url = urljoin("https://pretalx.com/api/v2", path)
+    resp = requests.get(
+        api_url,
+        headers={"Authorization": f"Token f32fb5b15e330c4441ebb0652976be345ba45c15"},
+    )
+    return resp.json()
+
+
+def get_talks(refetch=False):
+    selections_path = BASE_PATH / "talks.json"
+
+    def _relevant_session_type(talk):
+        if talk["submission_type"] == "Talk":
+            return True
+        return talk["submission_type"]["en"] in ["Workshop", "Keynote"]
+
+    if selections_path.exists() and refetch is False:
+        return json.load(selections_path.open())
+    else:
+        resp = api_get("events/djangocon-europe-2023/talks/?limit=50")
+        submissions = {
+            result["code"]: result
+            for result in resp["results"]
+            if _relevant_session_type(result)
+        }
+        with selections_path.open("w") as out:
+            json.dump(submissions, out)
+        return submissions
+
+
+def get_speakers(talk_codes, refetch=False):
+    speakers_path = BASE_PATH / "speakers.json"
+    if speakers_path.exists() and refetch is False:
+        return json.load(speakers_path.open())
+    else:
+        resp = api_get("events/djangocon-europe-2023/speakers/?limit=200")
+        speakers = {
+            result["code"]: result
+            for result in resp["results"]
+            if any(sub in talk_codes for sub in result["submissions"])
+        }
+        with speakers_path.open("w") as out:
+            json.dump(speakers, out)
+        return speakers
+
+
+def update_info(talk, speakers):
+    formatted = {
+        "title": talk["title"],
+        "slug": re.sub(r"\W+", "-", talk["title"]).strip("-").lower(),
+        "url": f"https://pretalx.com/djangocon-europe-2023/talk/{talk['code']}",
+        "type": talk["submission_type"]
+        if isinstance(talk["submission_type"], str)
+        else talk["submission_type"]["en"],
+        "state": talk["state"],
+        "speakers": [speaker["name"] for speaker in talk["speakers"]],
+        "slot": talk["slot"],
+    }
+    formatted["title"] = formatted["title"].replace("Keynote: ", "")
+    first_speaker = talk["speakers"][0]
+    formatted["photo_url"] = speakers[first_speaker["code"]]["avatar"]
+    formatted["name_slug"] = (
+        re.sub(r"\W+", "-", first_speaker["name"]).strip("-").lower()
+    )
+    return formatted
+
+
+def generate_session_md(session):
+    speakers = "\n".join([f"  - {speaker}" for speaker in session["speakers"]])
+    markdown = f"""---
+hidden: false
+layout: session-speaker-template
+speakers: 
+{speakers}
+permalink: /sessions/{session['name_slug']}/
+name_slug: {session['name_slug']}
```

Comment:
@rebkwok  I changed the name of file to the name of the first speaker slugified because I cannot update the shots.yml which is used to generate thumbnails. They are are all generated in `static/img/social/presenters`
If you think of a better way we can change but right now you should have all thumbnails.

Reply:
This doesn't quite work. I made the file name the slugified title because there are speakers with more than one talk/workshop - if you use the speaker name, you lose the first of each speaker's sessions (see cheuk and daniele - only their workshop session is there, but they both have a talk as well)

Reply:
Can we just add something to the slug if it's already been seen? We know there's only at most 2 per speaker, so just adding a "-1" should be fine